blocks/graph/__init__.py

```diff
@@ -104,7 +104,15 @@ def auxiliary_variables(self):
     @property
     def scan_variables(self):
         """Variables of Scan ops."""
-        return list(chain(*[g.variables for g in self._scan_graphs]))
+        # BFS
+        scan_graphs = self._scan_graphs
```

Comment:
You probably want to copy `scan_graphs` here, like e.g. `scan_graphs = list(self._scan_graphs)`.